### PR TITLE
feat(seed): scaffold BrewDog DIY Dog seed + Punk IPA 2007-2010 (1/25)

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

First slice of Issue #780. Ships the seed loader + type definitions + test harness for the 25-recipe BrewDog DIY Dog import, plus the historical **Punk IPA 2007-2010** as recipe #1. The remaining 24 recipes will land in follow-up PRs as each source page on `brewdogrecipes.com` is verified one at a time — keeps reviewer surface manageable and avoids ingesting unverified data in bulk.

## Architecture

- **New file** `packages/api/src/database/seeds/brewdog-diy-dog-recipes.seed.ts`
  - Types: `BrewdogDiyDogRecipeSeed`, `BrewdogDiyDogFermentableSeed`, `BrewdogDiyDogHopSeed`, `BrewdogDiyDogYeastSeed` — full sub-table breakdown for each recipe (vs `public-recipes.seed.ts` which carries metadata only).
  - Deterministic UUID range `00000000-0000-4001-8000-XXXXXXXXXXXX` (`-4001-` variant) so the existing 11 `-4000-` PUBLIC seed rows keep their identity.
  - Idempotent loader `seedBrewdogDiyDogRecipes(recipeRepo, fermentableRepo, hopRepo, yeastRepo, recipes?)` covering the Recipe row plus the three ingredient sub-tables.
- **Wipe-and-refill** pattern on the sub-tables: each run deletes rows by `recipe_id` then bulk-inserts the seed's ingredient lists. Deterministic post-condition regardless of drift between runs.
- `is_official` deliberately `false` on every backend row — same global-flag regression lesson as `public-recipes.seed.ts` (the matching service treats `is_official=true` as a 100-pt per-recipe boost, so tagging every DIY Dog row as official would surface them as "official" for unrelated scanned beers).
- `import_provenance` = `"BrewDog DIY Dog (open-source homebrew recipe book, attribution preserved)"` so the mobile UI's "Importée depuis…" badge can surface the source.

## Recipe #1 — Punk IPA 2007-2010

| Field | Value |
|---|---|
| UUID | `00000000-0000-4001-8000-000000000001` |
| Source | <https://brewdogrecipes.com/recipes/punk-ipa-2007-2010> |
| Style | American IPA |
| Batch | 20 L, 60 min boil |
| Metrics | OG 1.056 / FG 1.010 / ABV 6.0% / IBU 60 / EBC 17 |
| Fermentables | Extra Pale Malt 5.3 kg |
| Hops | 8 additions × Ahtanum / Chinook / Crystal / Motueka @ 60/15/5 min |
| Yeast | Wyeast 1056 American Ale @ 19°C primary |

**Coexistence with the existing `BrewDog DIY Dog Punk IPA`** at `-4000-...0b` (Issue #911 — the scan flow's "🏆 Recette officielle" row): the two are distinct millésimes useful for different audiences. The post-2010 row carries metadata only and serves the scan-flow Beat 4. The new 2007-2010 row carries the full DIY Dog book recipe — what users actually clone when they want to reproduce Punk IPA at home.

## Tests

8 specs covering happy / sad (idempotency) / edge paths:

- Insert path: every recipe + ingredient sub-rows seeded into empty tables.
- Visibility / owner / provenance / non-official invariants on every row.
- Idempotency: existing recipe rows updated in place, sub-tables wiped-and-refilled — verified via per-table delete call counts.
- Drifted-row reset: `visibility=PRIVATE`, `is_official=true`, `import_provenance="manually edited"` all forced back to canonical values on overwrite.
- Edge: deterministic UUID range matches `-4001-` variant, non-empty grain bill / hops / yeasts on every recipe, every recipe carries a `brewdogrecipes.com` source URL.

## Verification

- 662/664 API tests green (+8 new). 2 skipped + 1 suite skipped pre-existing.
- ESLint clean — the scan.service.ts:631 warning predates this PR.
- `nest build` clean.

## Out of scope

- Recipes 2-25 — separate top-up PRs as source data is verified.
- Runner script (`run-brewdog-diy-dog-seed.ts`) and orchestration with the existing `run-public-recipes-seed.ts` — deferred until at least 5 recipes ship so the runner has meaningful data to load.
- Mobile-side equivalent recipes mock updates — separate PR once enough recipes exist to define which DIY Dog row maps to which scanned bottle.

## Checklist

- [x] Happy path + sad path + edge cases covered (8 tests)
- [x] `nest build` clean
- [x] ESLint clean (no new warnings)
- [x] Existing tests still green (654 → 662 = +8 new)
- [x] No `any`, no inline styles, no hardcoded values introduced
- [x] Deterministic UUID range respected (`-4001-` variant)
- [x] Source URL on every recipe for traceability

Refs #780 (1/25 — scaffolding + first recipe; remaining 24 to follow)

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@vitalikevin — Pour info : on commence à charger le catalogue de recettes Brasse-Bouillon avec la première grosse référence brassicole, le **DIY Dog** de BrewDog (le livre de recettes maison qu'ils publient gratuitement). Cette PR pose les fondations (les types de données, le mécanisme de seed idempotent qui sait charger une recette + ses ingrédients dans la base) et ajoute la première recette : Punk IPA millésime 2007-2010.

La stratégie est d'ajouter ensuite les 24 autres recettes une à une dans des PR séparées, en vérifiant chaque page source (`brewdogrecipes.com`) avant de la committer — ça permet de revoir les données en petits lots sans noyer la review. Pour la démo de soutenance le 27 mai, ça remplit le hub Découvrir avec un catalogue brassicole crédible et reconnaissable.

Tous les checks CI passent (662/664 tests verts, build et lint propres).

File: packages/api/src/database/seeds/brewdog-diy-dog-recipes.seed.ts [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+      batch_size_l: recipe.batch_size_l,
+      boil_time_min: recipe.boil_time_min,
+      og_target: recipe.og_target,
+      fg_target: recipe.fg_target,
+      abv_estimated: recipe.abv_estimated,
+      ibu_target: recipe.ibu_target,
+      ebc_target: recipe.ebc_target,
+      efficiency_target: recipe.efficiency_target,
+      avg_rating: recipe.avg_rating,
+      brew_count: recipe.brew_count,
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/seeds/brewdog-diy-dog-recipes.seed.ts]: **Must Have — fixed in 258ceec.** `RecipeOrmEntity.brew_count` is `integer NOT NULL DEFAULT 0`. The loader now coalesces `recipe.brew_count ?? 0` so an omitted seed value lands as 0 rather than NULL. New regression spec pins it: `coalesces an omitted brew_count seed value to 0 rather than passing undefined to the non-nullable column`.

File: packages/api/src/database/seeds/brewdog-diy-dog-recipes.seed.ts [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+   * matching the `public-recipes.seed.ts` convention.
+   */
+  avg_rating?: number;
+  /**
+   * Number of times the recipe has been brewed by the community
+   * (synthetic value derived from BrewDog's Wisdom of the Crowd
+   * exposure). Same optionality contract as `avg_rating`.
+   */
+  brew_count?: number;
+  source_url: string;
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/src/database/seeds/brewdog-diy-dog-recipes.seed.ts:137]: **Must Have — fixed in 258ceec.** Replaced the global `BREWDOG_DIY_DOG_PROVENANCE` constant with a `buildBrewdogDiyDogProvenance(sourceUrl)` helper that bakes the per-recipe URL into `import_provenance` at seed time. The audit trail now reaches the database row (Punk IPA 2007-2010 stores e.g. `BrewDog DIY Dog (open-source homebrew recipe book, attribution preserved) — source: https://brewdogrecipes.com/recipes/punk-ipa-2007-2010`). Tests updated to assert the URL is present in every row's provenance and that the helper output matches the seed entry's source_url.
